hugegraph-server/hugegraph-dist/src/main/java/org/apache/hugegraph/dist/MemoryMonitor.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hugegraph.dist;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryUsage;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.hugegraph.config.HugeConfig;
+import org.apache.hugegraph.config.ServerOptions;
+import org.apache.hugegraph.util.Log;
+import org.slf4j.Logger;
+
+import com.sun.management.ThreadMXBean;
+
+public class MemoryMonitor {
+
+    private static final Logger LOG = Log.logger(MemoryMonitor.class);
+    private final double MEMORY_MONITOR_THRESHOLD;
+    private final int MEMORY_MONITOR_PERIOD;
+    private final ScheduledExecutorService scheduler;
+
+    public MemoryMonitor(String restServerConf) {
+        HugeConfig restServerConfig = new HugeConfig(restServerConf);
+        MEMORY_MONITOR_THRESHOLD =
+                restServerConfig.get(ServerOptions.JVM_MEMORY_MONITOR_THRESHOLD);
+        MEMORY_MONITOR_PERIOD =
+                restServerConfig.get(ServerOptions.JVM_MEMORY_MONITOR_PERIOD);
+        scheduler = Executors.newScheduledThreadPool(1);
```

Comment:
Maybe use `ExcecutorUtil.newScheduledThreadPool(1, "your thread name")`, to better manage threads